snapcraft/cli/_options.py

```diff
@@ -110,6 +111,19 @@ def get_build_environment(**kwargs):
     return env.BuilderEnvironmentConfig(force_provider=provider)
 
 
+def apply_host_build_environment_flags(build_environment_flags: Dict[str, str]) -> None:
```

Comment:
should this not live in cli.env?

Reply:
it can probably even be a method from the class that lives there, same for get_build_environment_flags

Reply:
I'm indifferent, but this is really an artifact that "host" is a special case provider and does not get a provider class.  Ideally we can refactor the provider bits to support launchpad and host as well...

My avoidance of cli.env / BuilderEnvironmentConfig is that it's blown away in: https://github.com/snapcore/snapcraft/pull/2678/files

Should I put it in env anyways?

Reply:
ah, forgot about that PR. Keep it here for now.